staging/src/k8s.io/client-go/transport/round_trippers.go

```diff
@@ -36,17 +36,18 @@ func HTTPWrappersForConfig(config *Config, rt http.RoundTripper) (http.RoundTrip
 	if config.WrapTransport != nil {
 		rt = config.WrapTransport(rt)
 	}
-
 	rt = DebugWrappers(rt)
-
 	// Set authentication wrappers
+	// TODO decide the priority here
 	switch {
 	case config.HasBasicAuth() && config.HasTokenAuth():
 		return nil, fmt.Errorf("username/password or bearer token may be set, but not both")
 	case config.HasTokenAuth():
 		rt = NewBearerAuthRoundTripper(config.BearerToken, rt)
 	case config.HasBasicAuth():
 		rt = NewBasicAuthRoundTripper(config.Username, config.Password, rt)
+	case config.HasTokenSource():
+		rt = NewTokenSourceRoundTripper(config.TokenSource, rt)
```

Comment:
I kind of expected this to be part of https://godoc.org/k8s.io/client-go/rest#RESTClient.Verb so we could bake this into the rest client's retry logic. e.g. when you hit a 401 you bust the cache.